ServerCore/Pages/Puzzles/Create.cshtml.cs

```diff
@@ -27,17 +28,21 @@ public IActionResult OnGet()
         [BindProperty]
         public Puzzle Puzzle { get; set; }
 
-        public async Task<IActionResult> OnPostAsync()
+        public async Task<IActionResult> OnPostAsync(int? eventid)
         {
             if (!ModelState.IsValid)
             {
                 return Page();
             }
 
+            if (eventid != null)
+            {
+                Puzzle.Event = await _context.Event.SingleOrDefaultAsync(m => m.ID == eventid);
+            }
             _context.Puzzle.Add(Puzzle);
             await _context.SaveChangesAsync();
 
-            return RedirectToPage("./Index");
+            return RedirectToPage("./Index", new { eventid = eventid });
```

Comment:
FYI @asyasky, take a look at how we should be doing this (but no need to block on this)